crates/pop-contracts/src/up.rs

```diff
@@ -106,6 +128,38 @@ pub async fn dry_run_gas_estimate_instantiate(
 	}
 }
 
+pub struct UploadDryRunResult {
+	pub result: String,
+	pub code_hash: String,
+	pub deposit: String,
+}
+
+/// Performs a dry-run for uploading a contract without modifying the state of the blockchain.
+///
+/// # Arguments
+///
+/// * `upload_exec` - the preprocessed data to upload a contract.
+///
+pub async fn dry_run_upload(
+	upload_exec: &UploadExec<DefaultConfig, DefaultEnvironment, Keypair>,
+) -> Result<UploadDryRunResult, Error> {
+	match upload_exec.upload_code_rpc().await? {
+		Ok(result) => {
+			let upload_result = UploadDryRunResult {
+				result: String::from("Success!"),
```

Comment:
Why the need for this? Especially as a string type. The fact that the function returns a Result should be enough to determine that it was a success. Please remove this unnecessary field.

Reply:
True, no need.